uaclient/files/files.py

```diff
@@ -64,6 +64,36 @@ def delete(self):
         system.ensure_file_absent(self.path)
 
 
+class ProJSONFile:
+    def __init__(
+        self,
+        name: str,
+        directory: str = defaults.DEFAULT_DATA_DIR,
+        private: bool = True,
+    ):
+        self._file = UAFile(name=name, directory=directory, private=private)
```

Comment:
You could accept a `UAFile` as an argument instead of drilling arguments
```suggestion
    def __init__(
        self,
        ua-file: UAFile,
    ):
        self._file = ua_file
```